packages/docusaurus/src/server/configValidation.ts

```diff
@@ -152,17 +152,25 @@ const I18N_CONFIG_SCHEMA = Joi.object<I18nConfig>({
   .optional()
   .default(DEFAULT_I18N_CONFIG);
 
-const SiteUrlSchema = URISchema.required().custom((value: string, helpers) => {
-  try {
-    const {pathname} = new URL(String(value));
-    if (pathname !== '/') {
-      helpers.warn('docusaurus.configValidationWarning', {
-        warningMessage: `The url is not supposed to contain a sub-path like '${pathname}'. Please use the baseUrl field for sub-paths.`,
-      });
+const SiteUrlSchema = Joi.string()
+  .required()
+  .custom((value: string, helpers) => {
+    try {
+      const {pathname} = new URL(value);
+      if (pathname !== '/') {
+        helpers.warn('docusaurus.configValidationWarning', {
```

Comment:
Thanks

in case you want to follow-up, I think we should make this an error instead of a warning

Reply:
Thanks for the suggestion! I'll work on this and make a new PR.